kclvm/tools/src/LSP/src/dispatcher.rs

```diff
@@ -121,9 +123,16 @@ impl<'a> RequestDispatcher<'a> {
 
             move || {
                 let result = compute_response_fn(snapshot, params, sender.clone());
-                sender
-                    .send(Task::Response(result_to_response::<R>(id, result)))
-                    .unwrap();
+                match &result {
+                    Err(e) if e.to_string() == RETRY_REQUEST => {
```

Comment:
I suggest using `this-error` lib to define errors instead of hard coding strings to classify errors, as this will increase the cost of judgment and require minimizing overhead in the language server as much as possible.